sympy/core/function.py

```diff
@@ -2276,9 +2276,13 @@ def doit(self, **hints):
                 # do Subs that aren't related to differentiation
                 undone2 = []
                 D = Dummy()
+                arg = e.args[0]
+                arg = repr(arg)
                 for vi, pi in undone:
                     if D not in e.xreplace({vi: D}).free_symbols:
-                        e = e.subs(vi, pi)
+                        vrbl = repr(vi)
+                        if arg.find(vrbl)!=-1:
+                            e = e.subs(vi, pi)
```

Comment:
This is calling `arg.find` with a string. I'm not sure what that's supposed to do but we should convert to string as part of internal operations.

Reply:
vrbl stores the variable w.r.t. which we are differentiating(x in my e.g.). arg.find(vrbl) returns **-1**, if there is no vi(x in e.g.) in the argument of e, thus this means we don't need to use subs method on e as there is not vi present in e. Thus there will be no incorrect substitution.

Reply:
Can this be `if arg.has(vi)`?

Reply:
Yes, this just works the best, Should this be implemented instead of strings?

Reply:
Something like that. We shouldn't use strings.

Reply:
Thank you for the suggestion, I made the necessary changes and it looks more compact and quick.

Please review and kindly update if further betterment can be done.
Thanks a lot!